### ISSUE TITLE
Display label for additional fonts

### ISSUE THREAD

**Issue by andreiglingeanu** [elementor/elementor#12822]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

I've seen [here](https://github.com/elementor/elementor/issues/5089), [here](https://github.com/elementor/elementor/issues/5438) and [here](https://github.com/elementor/elementor/issues/4624) a lot of things written down on how to use the `elementor/fonts/additional_fonts` filter for adding additional fonts to Elementor.

It all works great, except you can't define a separate display name for the font (to be showed in the Elementor UI) and another value to be the actual string used in the `font-family: <font_name>` CSS rule. Right now these have to match.

**Describe the solution you'd like**

I'd expect `elementor/fonts/additional_fonts` to allow us to define a display label **and** a font name that should be used in CSS.

I realize stuff like this should be implemented right from the beginning and can be hard to introduce at this point, especially if there's no big need for it.

**Describe alternatives you've considered**
The alternative is to just settle with the slightly degraded label in Elementor's UI in some cases. For ex. Typekit's CSS returns font families with dashes and there's no way to change that.

Or, if that's feasible too, will happily hook into the JS of the font family picker to change the labels there.

Happy to contribute back a solution but will need some guidance on what will be the best way to go about this.

cc. @bainternet

**Comment by DorShahar** [comment]
Hi @andreiglingeanu 
Please see this code snippet from [this link](https://github.com/elementor/elementor/issues/5089) which you posted yourself: 
```
/**
 * Add Group Fonts
 */
add_filter( 'elementor/fonts/additional_fonts', function( $additional_fonts ) {
	// Key/value
	//Font name/font group
	$additional_fonts['MyFont Font Family Name'] = 'my_fonts';
	$additional_fonts['MyFont Font Family Name2'] = 'my_fonts';
	return $additional_fonts;
} );
```


- Where it says `MyFont Font Family Name` will determine what you will see Elementor's UI font family typography dropdown (this has to be equal to `font-family` in your CSS file). 
- To determine the `font-family` name that will be used in the CSS, it should be done in CSS file you used to enqueue your font, using `@font-face` under the `font-family` property. 

```
add_action( 'elementor/fonts/print_font_links/my_fonts', function( $font_name ) {
  // Enqueue your font
echo '<link rel="stylesheet" href="my-font-face-stylesheet.css" />';
} );
```
```
// my-font-face-stylesheet.css
@font-face {
  font-family: "MyFont Font Family Name";
  src: url("/fonts/OpenSans-Regular-webfont.woff2") format("woff2"),
       url("/fonts/OpenSans-Regular-webfont.woff") format("woff");
}

@font-face {
  font-family: "MyFont Font Family Name 2";
.
.
.
}
```

**Comment by andreiglingeanu** [comment]
@DorShahar Thanks a lot for your reply!

Yes, but then Elementor will always put `MyFont Font Family Name` in `font-family` rules.

We really need this:

Put `MyFont Font Family Name` in Elementor's dropdown UI, but have it like this in the font family rule: `font-family: myfont-font-family-name`. I'm aware that I'll still responsible for correctly outputting `@font-face` rules (and they'll be referencing the font as `myfont-font-family-name`).

Hope this makes sense and please let me know if it doesn't, I'll try to provide a clearer explanation.

**Comment by DorShahar** [comment]
Hi @andreiglingeanu , I got your point.
Unfortunately, this is not supported. But I'll mark it as a request for us.

**Comment by andreiglingeanu** [comment]
Thanks @DorShahar! This is going to be really useful with font sources like Typekit, since they provide font families with dashes in their CSS.

**Comment by prashantcs8** [comment]
This is a bit tricky but doable, 
for example I added "danielblack" custom font
Step 1 created fonts folder in the child theme folder and paste font files (.eot, .woff, .ttf).
Step 2 in the child theme folder add css in style.css
```
@font-face {
    font-family: 'danielblack';
    src: url('Daniel-Black-webfont.eot');
    src: url('fonts/Daniel-Black-webfont.eot?#iefix') format('embedded-opentype'),       
         url('fonts/Daniel-Black-webfont.woff') format('woff'),
         url('fonts/Daniel-Black-webfont.ttf') format('truetype');       
    font-weight: normal;
    font-style: normal;

}
```
Step 3 add code in functions.php
```
/**
 * Add Font Group to Elementor
 */
add_filter( 'elementor/fonts/groups', 'prefix_elementor_custom_fonts_group', 10, 1 );
function prefix_elementor_custom_fonts_group( $font_groups ) {

	$font_groups['danielblack'] = __( 'Daniel Black' );
	return $font_groups;
}
/**
 * Add Group Fonts to Elementor
 */
add_filter( 'elementor/fonts/additional_fonts', 'prefix_elementor_custom_fonts', 10, 1 );
function prefix_elementor_custom_fonts( $additional_fonts ) {
	// Key/value
	//Font name/font group
	$additional_fonts['danielblack'] = 'danielblack';
	return $additional_fonts;
}
```
then you will find "Daniel Black" group "danielblack" fonts in Typography > fonts dropdown

**Comment by andreiglingeanu** [comment]
@danielkatz Yes, but can we add multiple fonts with the right labels in the same group using this method?

**Comment by prashantcs8** [comment]
Yes It Works fine
For Example, I added 2 more variants of the same font
in the child theme folder add css in style.css
```
@font-face {
    font-family: 'danielblack';
    src: url('fonts/Daniel-Black-webfont.eot');
    src: url('fonts/Daniel-Black-webfont.eot?#iefix') format('embedded-opentype'),         
         url('fonts/Daniel-Black-webfont.woff') format('woff'),
         url('fonts/Daniel-Black-webfont.ttf') format('truetype'),
         url('fonts/Daniel-Black-webfont.svg#danielblack') format('svg');
    font-weight: normal;
    font-style: normal;

}
@font-face {
    font-family: 'danielbold';
    src: url('fonts/Daniel-Bold-webfont.eot');
    src: url('fonts/Daniel-Bold-webfont.eot?#iefix') format('embedded-opentype'),         
         url('fonts/Daniel-Bold-webfont.woff') format('woff'),
         url('fonts/Daniel-Bold-webfont.ttf') format('truetype'),
         url('fonts/Daniel-Bold-webfont.svg#danielbold') format('svg');
    font-weight: normal;
    font-style: normal;

}
@font-face {
    font-family: 'danielregular';
    src: url('fonts/Daniel-Regular-webfont.eot');
    src: url('fonts/Daniel-Regular-webfont.eot?#iefix') format('embedded-opentype'),         
         url('fonts/Daniel-Regular-webfont.woff') format('woff'),
         url('fonts/Daniel-Regular-webfont.ttf') format('truetype'),
         url('fonts/Daniel-Regular-webfont.svg#danielregular') format('svg');
    font-weight: normal;
    font-style: normal;

}
```
in functions.php
```

/**
 * Add Font Group to Elementor
 */
add_filter( 'elementor/fonts/groups', 'prefix_elementor_custom_fonts_group', 10, 1 );
function prefix_elementor_custom_fonts_group( $font_groups ) {

	$font_groups['danielblack'] = __( 'Daniel Black' );
	return $font_groups;
}
/**
 * Add Group Fonts to Elementor
 */
add_filter( 'elementor/fonts/additional_fonts', 'prefix_elementor_custom_fonts', 10, 1 );
function prefix_elementor_custom_fonts( $additional_fonts ) {
	// Key/value
	//Font name/font group
	$additional_fonts['danielblack'] = 'danielblack';
	$additional_fonts['danielbold'] = 'danielblack';
	$additional_fonts['danielregular'] = 'danielblack';
	return $additional_fonts;
}
```

**Comment by andreiglingeanu** [comment]
@prashantcs8 Yes, but this isn't quite it to be frank. I knot you can add multiple groups there. But I'm looking for a way to add 1 group and multiple fonts in it, with different display label: https://cln.sh/9blmw6

As of the current implementation, this is not possible to do in Elementor unfortunately.

**Comment by prashantcs8** [comment]
Yes It's possible @andreiglingeanu please try my above snippet
https://imgur.com/a/JMPbZbM

**Comment by javiercriado** [comment]
> /**
>  * Add Font Group to Elementor
>  */
> add_filter( 'elementor/fonts/groups', 'prefix_elementor_custom_fonts_group', 10, 1 );
> function prefix_elementor_custom_fonts_group( $font_groups ) {
> 
> 	$font_groups['danielblack'] = __( 'Daniel Black' );
> 	return $font_groups;
> }
> /**
>  * Add Group Fonts to Elementor
>  */
> add_filter( 'elementor/fonts/additional_fonts', 'prefix_elementor_custom_fonts', 10, 1 );
> function prefix_elementor_custom_fonts( $additional_fonts ) {
> 	// Key/value
> 	//Font name/font group
> 	$additional_fonts['danielblack'] = 'danielblack';
> 	$additional_fonts['danielbold'] = 'danielblack';
> 	$additional_fonts['danielregular'] = 'danielblack';
> 	return $additional_fonts;
> }

thanks a lot!

In my case I wanted to use a google font which is new and it's not in the fonts.php list.

so I created a "google 2" font group in functions.php file and imported the font in my child theme style.css like this

`<style>
@import url('https://fonts.googleapis.com/css2?family=Rubik+Glitch&display=swap');
</style>

font-family: 'Rubik Glitch', cursive;
`
